### PR TITLE
Enable non unicode character in catalog import.

### LINKED ISSUE #4276: catalog-import plugin does not support Unicode characters in catalog-info.yaml file
## Expected Behavior
Importing a GitHub repo should allow Unicode characters in the submitted catalog-info.yaml file. For example, a file like this should work:
```YAML
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: test-service
  description: Description with emojis 🤖
spec:
  lifecycle: production
  type: service
  owner: test-team
```

## Current Behavior
Right now if you try to import the file above, you will get the following error: `Failed to submit PR to repo, Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.`

## Possible Solution
Use a library like [js-base64](https://www.npmjs.com/package/js-base64) to encode the file contents, which allows for Unicode characters.

## Steps to Reproduce
1. Create a custom LocationAnalyzer
2. Modify the generated Entity in the custom LocationAnalyzer by adding a description field with Unicode characters in it
3. Use custom LocationAnalyzer
4. Try to import GitHub repository via `/catalog-import`

## Context
We are trying to add some extra information to the generated catalog-info.yaml file. This reduces the work for developers as they can import the file and not make any modifications afterwards. We fetch the description from GitHub and add it to the generated catalog-info.yaml file. Some of our repos have emojis in them, so creating the PR fails due to the lack of Unicode support in the native `btoa` function.

### PR DESCRIPTION
**Pull request by padraigobrien**
replaced btoa with js-base64 so the code can gracefully handle non unicode characters.

Signed-off-by: Podge <padraig@padraigobrien.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Fixes #4276

### CODE REVIEW

File: .changeset/three-icons-smell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-import': patch
+---
+
+gh-4276, When importing components you will now have the ability to use non unicode characters in the entity owner field
```
Comment (benjdlambert) [inline .changeset/three-icons-smell.md:5]: Think we can drop the issue number from here to keep them looking nice in the CHANGELOGs :)

**Comment by benjdlambert** [comment]
Closes #4276

**Comment by benjdlambert** [comment]
sorry @padraigobrien last go, can you rebase from master again. we had a problem with the e2e tests late last evening

**Comment by padraigobrien** [comment]
did a git rebase --signoff HEAD~3 to amend my last three commits

**Comment by padraigobrien** [comment]
Think it did not have the intended effect

**Comment by padraigobrien** [comment]
I'm going to close this pull request and start a fresh one

**Pull request closed without merging**